### PR TITLE
[WIP] remove the use of param_find("")

### PR DESCRIPTION
**Pull request by BazookaJoe1900**
param_find() does binary search every time any other parameter is beeing changed. which is extensive  CPU process.
This commit doesn't compile, because of calling non existing parameters FW_MAN_R_SC, FW_MAN_P_SC, FW_MAN_Y_SC

**Describe problem solved by the proposed pull request**
described on: 'extensive on every parameter change #12992'

The current problem is that do_trim_calibration() can be called on systems that doesn't has FW_MAN_* parameters. and in the new method, that does not compile

### CODE REVIEW

File: src/modules/commander/rc_calibration.cpp [hunk 1]
```diff
@@ -65,34 +66,37 @@ int do_trim_calibration(orb_advert_t *mavlink_log_pub)
...
-	float yaw_trim_active;
-	param_get(param_find("TRIM_YAW"), &yaw_trim_active);
+	param_t param_trim_roll = param_handle(px4::params::TRIM_ROLL);
+	param_t param_trim_pitch = param_handle(px4::params::TRIM_PITCH);
+	param_t param_trim_yaw = param_handle(px4::params::TRIM_YAW);
+	param_t param_fw_man_r_sc = param_handle(px4::params::FW_MAN_R_SC);
+	param_t param_fw_man_p_sc = param_handle(px4::params::FW_MAN_P_SC);
+	param_t param_fw_man_y_sc = param_handle(px4::params::FW_MAN_Y_SC);
+
+	float roll_trim_active, pitch_trim_active, yaw_trim_active, roll_scale, pitch_scale, yaw_scale;
```
Comment (dagar) [inline src/modules/commander/rc_calibration.cpp]: It's probably a bit pedantic here, but I think it's good practice to declare and initialize each of these on a separate line with a sane default value.

**Comment by dagar** [comment]
Cool, I forgot we had this now. Thanks for the reminder. https://github.com/PX4/Firmware/pull/13081

File: src/modules/commander/rc_calibration.cpp (hunk 1 above)
Comment (BazookaJoe1900) [reply src/modules/commander/rc_calibration.cpp]: @dagar the issue here is not mainly about the searching of the parameter, but about the problem that the rc_calibration search for parameter that doesn't exists, on some compilation

File: src/drivers/uavcan/uavcan_main.cpp [hunk 2]
```diff
@@ -420,18 +421,8 @@ int  UavcanNode::get_param(int remote_node_id, const char *name)
...
-	if (param_handle != PARAM_INVALID) {
-		param_get(param_handle, &_airmode);
-	}
-
-	param_handle = param_find("THR_MDL_FAC");
-
-	if (param_handle != PARAM_INVALID) {
-		param_get(param_handle, &_thr_mdl_factor);
-	}
+	param_get(param_handle(px4::params::MC_AIRMODE), &_airmode);
```
Comment (dagar) [inline src/drivers/uavcan/uavcan_main.cpp]: MC_AIRMODE needs to be optional for UAVCAN.

**Comment by dagar** [comment]
Nice work, this saves ~8.7KB of flash on px4_fmu-v5_default.

http://ci.px4.io:8080/blue/organizations/jenkins/PX4%2FFirmware/detail/PR-13076/4/pipeline/25#step-387-log-114

File: src/modules/fw_att_control/FixedwingAttitudeControl.cpp [hunk 3]
```diff
@@ -380,7 +381,7 @@ FixedwingAttitudeControl::vehicle_status_poll()
 
 				int32_t vt_type = -1;
 
-				if (param_get(param_find("VT_TYPE"), &vt_type) == PX4_OK) {
+				if (param_get(param_handle(px4::params::VT_TYPE), &vt_type) == PX4_OK) {
```
Comment (dagar) [inline src/modules/fw_att_control/FixedwingAttitudeControl.cpp]: VTOL needs to be optional for FW & MC controllers.

Comment (BazookaJoe1900) [reply src/modules/fw_att_control/FixedwingAttitudeControl.cpp]: there are many issues about mixing VTOL/Rover/Multicopter/Fixwing

**Comment by BazookaJoe1900** [comment]
closed in favor to reduce the use of param_find("XXX") #13089

**Pull request closed without merging**